### ISSUE TITLE
Hello,Is there specific hyperparameter configuration for cifar10 and cifar100?

### ISSUE THREAD

**Issue by Lutong-Qin** [LeapLabTHU/L2W-DEN#1]
Thank you for your answer

**Comment by yizenghan** [comment]
Hi, we will polish the code and provide the CIFAR training configuration soon (after CVPR deadline).

**Comment by yifanpu001** [comment]
The detailed configuration of CIFAR has been illustrated in the paper. We reorganized it here:

For the classification model architecture, we follow the original MSDNet design, which is
> --arch msdnet --nBlocks 5 --stepmode lin_grow --step 1 --base 1 --nChannels 16 \

For the meta net, as described in Sec. 3.2 and Appendix C, it has one hidden layer, and its width is 500 (for a 5-exit model). The interval I of updating WPN in Algorithm 1 is set as 1 for CIFAR, with an initial learning rate 1e-4 and a weight decay of 1e-4 (Appendix A.4).
> --meta_net_hidden_size 500 --meta_net_num_layers 1 --meta_interval 1 --meta_lr 1e-4 --meta_weight_decay 1e-4 \

The magnitude of perturbation δ is set as 0.8 on CIFAR, while the budget controlling variable q is set as 0.75(Appendix A.3). (q = target_p_index / 20). We use CE loss as WPN's input (Sec. 4.1). The normalization dimension hyperparameter '--constraint_dimension'  is set as 'col'.
> --epsilon 0.8 --target_p_index 15 --meta_net_input_type loss --constraint_dimension col \

For the training configuration, we also follow the config in MSDNet without any change.
> --epochs 300 --batch-size 1024 --lr 0.8 --lr-type cosine --print-freq 10;

As a result, the training script for a 5-exits MSDNet on CIFAR100 is
> python tools/main_cifar.py \
> --train_url YOUR_SAVE_PATH \
> --data_url YOUR_DATA_PATH --workers 1 --seed 1 \
> --arch msdnet --nBlocks 5 --stepmode lin_grow --step 1 --base 1 --nChannels 16 \
> --meta_net_hidden_size 500 --meta_net_num_layers 1 --meta_interval 1 --meta_lr 1e-4 --meta_weight_decay 1e-4 \
> --epsilon 0.8 --target_p_index 15 --meta_net_input_type loss --constraint_dimension col \
> --epochs 300 --batch-size 1024 --lr 0.8 --lr-type cosine --print-freq 10;

**Comment by metodj** [comment]
Hey,

thanks a lot for the nice codebase. 

I wanted to rerun some of your CIFAR experiments, however, I noticed that the `main_cifar.py` file you are mentioning in this thread is missing. Could you upload the file? 

Thanks in advance!
